agent/consul/state/catalog.go

```diff
@@ -1086,6 +1086,43 @@ func (s *Store) EnsureCheck(idx uint64, hc *structs.HealthCheck) error {
 	return nil
 }
 
+// This ensure all connected services to a check are updated properly
+func (s *Store) alterServicesFromCheck(tx *memdb.Txn, idx uint64, hc *structs.HealthCheck, checkServiceExists bool) error {
+	// If the check is associated with a service, check that we have
+	// a registration for the service.
+	if hc.ServiceID != "" {
+		service, err := tx.First("services", "id", hc.Node, hc.ServiceID)
+		// When deleting a service, service might have been removed already
```

Comment:
I'm confused, I guess this is refering to the `checkServiceExists` flag passed which seems reasonable, but then in the case that we _don't_ check existence and so skip the first error handler, the second handler will be triggered and return an error anyway?

The caller in `deleteCheckTxn` doesn't distinguish between those error types so will fail if this returns any error anyway. Can we just remove this case as it seems to do nothing?

Reply:
Ok, I did it another way in next patch: I renamed alterServicesFromCheck and it only updates all Service Indexes for a given node